pymongo/encryption.py

```diff
@@ -552,6 +554,95 @@ def __init__(
+                }
+
+          - ``encryptedFields`` (optional) (dict): **(BETA)** Document that describes the encrypted fields for
+            Queryable Encryption. For example::
+
+                {
+                  "escCollection": "enxcol_.encryptedCollection.esc",
+                  "eccCollection": "enxcol_.encryptedCollection.ecc",
+                  "ecocCollection": "enxcol_.encryptedCollection.ecoc",
+                  "fields": [
+                      {
+                          "path": "firstName",
+                          "keyId": Binary.from_uuid(UUID('00000000-0000-0000-0000-000000000000')),
+                          "bsonType": "string",
+                          "queries": {"queryType": "equality"}
+                      },
+                      {
+                          "path": "ssn",
+                          "keyId": Binary.from_uuid(UUID('04104104-1041-0410-4104-104104104104')),
+                          "bsonType": "string"
+                      }
+                    ]
+                }
+
+            The "keyId" may be set to `None` to auto-generate the data keys.
+          - `**kwargs` (optional): additional keyword arguments are the same as "create_collection".
+
+        All optional `create collection command`_ parameters should be passed
+        as keyword arguments to this method.
+        See the documentation for :meth:`~pymongo.database.Database.create_collection` for all valid options.
+
+        .. versionadded:: 4.4
+           Added the codec_options, read_preference, and write_concern options.
+
+        .. _create collection command:
+            https://mongodb.com/docs/manual/reference/command/create
+
+        """
+        if not data_key_opts:
+            data_key_opts = {}
+        encrypted_fields = database._get_encrypted_fields(kwargs, name, False)
+        if encrypted_fields:
+            for field in encrypted_fields["fields"]:
+                if isinstance(field, dict) and field.get("keyId") is None:
+                    try:
+                        field["keyId"] = self.create_data_key(
+                            kms_provider=kms_provider,  # type:ignore[arg-type]
+                            **data_key_opts,
+                        )
+                    except EncryptionError as exc:
+                        raise EncryptionError(
+                            Exception(
+                                f"Error occurred while creating data key for field {field['path']} with encryptedFields={str(encrypted_fields)}"
+                            )
+                        ) from exc
+            kwargs["encryptedFields"] = encrypted_fields
+        else:
+            raise ConfigurationError(
+                "'encryptedFields' must be provided as a keyword argument to create_encrypted_collection"
+                "or initialized with AutoEncryptionOpts."
+            )
+
+        return database.create_collection(name=name, **kwargs), encrypted_fields
```

Comment:
New requirement just came in to return the encrypted_fields when create fails as well: https://github.com/mongodb/specifications/pull/1375

Reply:
Done.